### PR TITLE
Feat/remove timeout from detectors

### PR DESCRIPTION
**Pull request by pitoniak32**
related: https://github.com/open-telemetry/opentelemetry-rust/issues/2320, https://github.com/open-telemetry/opentelemetry-rust/pull/2322
## Changes

Remove the `ResourceDetector.detect()` timeout option.

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/resource/mod.rs [hunk 1]
```diff
@@ -262,13 +258,12 @@ pub trait ResourceDetector {
     ///
     /// If source information to construct a Resource is invalid, for example,
     /// missing required values. an empty Resource should be returned.
-    fn detect(&self, timeout: Duration) -> Resource;
+    fn detect(&self) -> Resource;
```
Comment (cijothomas) [inline opentelemetry-sdk/src/resource/mod.rs:261]: can we keep it obsolete for 0.27.1 release and remove in 0.28?

Comment (pitoniak32) [reply opentelemetry-sdk/src/resource/mod.rs:261]: Yup! I can update it tomorrow 👍

Comment (pitoniak32) [reply opentelemetry-sdk/src/resource/mod.rs:261]: is there a clean way of marking one argument as deprecated? from what I can tell we wont be able to allow users to remove the usage even if it gets marked as deprecated since we would need to give a new function on the Detector trait

Comment (cijothomas) [reply opentelemetry-sdk/src/resource/mod.rs:261]: I am not able to find a way to do that in Rust! Lets hold on to this and merge 0.27.1 is out (~1 week)

**Review by cijothomas — CHANGES_REQUESTED** [review]
Blocking accidental merge until https://github.com/open-telemetry/opentelemetry-rust/pull/2332#discussion_r1855475107

File: opentelemetry-sdk/CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@
 Released 2024-Nov-27
 
 - **DEPRECATED**:
+  - `ResourceDetector.detect()` no longer supports timeout option.
```
Comment (cijothomas) [inline opentelemetry-sdk/CHANGELOG.md:16]: need to move this to be under vnext.

File: stress/src/throughput.rs [hunk 3]
```diff
@@ -155,7 +155,7 @@ impl<'a> UnsafeSlice<'a> {
     #[inline(always)]
     unsafe fn increment(&self, i: usize) {
         let value = self.slice[i].get();
-        (*value).count = (*value).count + 1;
+        (*value).count += 1;
```
Comment (cijothomas) [inline stress/src/throughput.rs:158]: nit: Please remove unrelated changes.

**Review by cijothomas — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

File: stress/src/throughput.rs (hunk 3 above)
Comment (pitoniak32) [reply stress/src/throughput.rs]: removed
